src/main/java/com/internet/shop/dao/UserDaoJdbcImpl.java

```diff
@@ -0,0 +1,232 @@
+package com.internet.shop.dao;
+
+import com.internet.shop.exceptions.DataProcessingException;
+import com.internet.shop.lib.Dao;
+import com.internet.shop.models.Role;
+import com.internet.shop.models.User;
+import com.internet.shop.util.ConnectionUtil;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+@Dao
+public class UserDaoJdbcImpl implements UserDao {
+    @Override
+    public Optional<User> findByLogin(String login) {
```

Comment:
this method is too long, move some code to private methods

Reply:
fixed